### ISSUE TITLE
Robustify Gitlab CI configuration

### ISSUE THREAD

**Issue by rhornung67** [LLNL/RAJA#984]
To make our LC Gitlab CI configuration yield increased throughput and avoid failures due to timeouts, disk quota exceeded, etc.:

- [x] For TOSS3 testing, move from quartz to ruby. Consider using batch allocations rather than debug.
- [x] Examine how we are using resources for builds and tests and correct as needed to avoid running too many builds in parallel on a single node, avoid running tests (esp. OpenMP) in parallel with builds and other tests to avoid thrashing, etc.
- [x] Remove old build/test files (only keep one for each active branch, if possible, and delete older than some time interval) or from CI checks that pass to avoid exceeding disk quota limits. Ideally, this can be done in Gitlab tool. Otherwise, develop scripts for this.
- [x] Explore options for cloning repos into node local storage that is not on the files system.

**Comment by rhornung67** [comment]
It seems we have completed these tasks. We can add new issue to address other related concerns
